### PR TITLE
feat: add learners enrolment list api

### PR DESCRIPTION
**Pull request by tehreem-sadat**
Add enrollment api.

**Example usage:**

To get all enrolments:

`/api/fx/learners/v1/enrollments/`

```
[ 
    {
            "user_id": 10,
            "full_name": "user1",
            "alternative_full_name": "عالي",
            "username": "user1",
            "national_id": "",
            "email": "[user1@example.com](mailto:user1@example.com)",
            "mobile_no": null,
            "year_of_birth": null,
            "gender": "",
            "gender_display": null,
            "date_joined": "2024-11-08T12:31:06.322903Z",
            "last_login": "2024-11-15T08:23:03.539659Z",
            "certificate_available": false,
            "course_score": null,
            "active_in_course": false,
            "course_id": "course-v1:bragi_org+111+11"
        },
]
...
```

To get all enrolments - with complete details

`/api/fx/learners/v1/enrollments/?optional_field_tags=__all__`

```
[
    {
            "user_id": 10,
            "full_name": "user1",
            "alternative_full_name": "عالي",
            "username": "user1",
            "national_id": "",
            "email": "[user1@example.com](mailto:user1@example.com)",
            "mobile_no": null,
            "year_of_birth": null,
            "gender": "",
            "gender_display": null,
            "date_joined": "2024-11-08T12:31:06.322903Z",
            "last_login": "2024-11-15T08:23:03.539659Z",
            "certificate_available": false,
            "course_score": null,
            "active_in_course": false,
            "progress": 0.0,
            "certificate_url": null,
            "course_id": "course-v1:bragi_org+111+11",
            "earned - Final Exam: Subsection": "no attempt",
            "possible - Final Exam: Subsection": "no attempt",
            "earned - another: Subsection": "no attempt",
            "possible - another: Subsection": "no attempt"
    }.
...
]
```

To get all enrolments - for certain courses -> Send ',' separated list of course ids:

`/api/fx/learners/v1/enrollments/?course_ids=course_id1, course_id2`

To get all enrolments - for certain users -> Send ',' separated list of user ids:

`/api/fx/learners/v1/enrollments/?user_ids=1,2`

To get all enrolments - for specific user and course -> Send ',' separated list of user ids and course_ids:

`/api/fx/learners/v1/enrollments/?user_ids=1,2&course_ids=course_id1`

**Note**:
500 error will be raised for invalid course_ids.

### CODE REVIEW

File: futurex_openedx_extensions/dashboard/serializers.py [hunk 1]
```diff
@@ -340,9 +342,63 @@ def _extract_exam_scores(representation_item: dict[str, Any]) -> None:
...
+        """Get the User. Its helper method required for CourseScoreAndCertificateSerializer"""
+        return obj.user if obj else None
+
+    def get_course_id(self, obj: Any) -> str:
+        """Get course id"""
+        return str(self._get_course_id(obj))
+
+    def to_representation(self, instance: Any) -> Any:
+        representation = LearnerBasicDetailsSerializer(self._get_user(instance)).data
+        representation.update(super().to_representation(instance))
```
Comment (shadinaif) [inline futurex_openedx_extensions/dashboard/serializers.py:401]: why do we merge two serializers here like this? can't we add `LearnerBasicDetailsSerializer` as an inherited parent?

**Review by shadinaif — CHANGES_REQUESTED** [review]

File: futurex_openedx_extensions/dashboard/serializers.py (hunk 1 above)
Comment (tehreem-sadat) [reply futurex_openedx_extensions/dashboard/serializers.py:401]: I tried to do this way initially, but models are different for both serializers so I was getting issues, I'll try to look into it one more time.

File: futurex_openedx_extensions/dashboard/serializers.py [hunk 2]
```diff
@@ -166,20 +177,36 @@ def _get_arabic_name(self, obj: Any) -> str:
...
+        """Return user ID."""
+        return self._get_user(obj).username  # type: ignore
+
+    def get_date_joined(self, obj: get_user_model) -> str:
+        """Return user ID."""
+        return str(self._get_user(obj).date_joined)  # type: ignore
+
+    def get_last_login(self, obj: get_user_model) -> str:
+        """Return user ID."""
+        return str(self._get_user(obj).last_login)  # type: ignore
```
Comment (shadinaif) [inline futurex_openedx_extensions/dashboard/serializers.py:208]: don't trust `str` when converting date and time. At least it is not timezone-aware
```suggestion
    def get_date_joined(self, obj: Any) -> str:
        date_joined = self._get_user(obj).date_joined
        return date_joined.isoformat() if date_joined else None

    def get_last_login(self, obj: Any) -> str:
        last_login = self._get_user(obj).last_login
        return last_login.isoformat() if last_login else None
```

File: futurex_openedx_extensions/dashboard/serializers.py [hunk 3]
```diff
@@ -340,12 +347,91 @@ def _extract_exam_scores(representation_item: dict[str, Any]) -> None:
                 representation_item[possible_key] = score[1] if score else 'no attempt'
 
         representation = super().to_representation(instance)
-
```
Comment (shadinaif) [inline futurex_openedx_extensions/dashboard/serializers.py:343]: please recover the deleted empty line, here and everywhere unless you intentionally want to delete it

For this one, it is more readable to have the empty lines separating the logic

**Review by shadinaif — APPROVED** [review]
great work @tehreem-sadat !

please address my comment about `str` then it's good to go 👍🏼

File: futurex_openedx_extensions/dashboard/serializers.py (hunk 1 above)
Comment (shadinaif) [reply futurex_openedx_extensions/dashboard/serializers.py]: thank you for changing the approach in the last push 🌷

**Pull request merged**